### ISSUE TITLE
Crash when list updated during drag action

### ISSUE THREAD

**Issue by leo-k** [bauerca/drag-sort-listview#7]
Updating the list, using setAdapter, while the user is in the middle of dragging leads to an exception being thrown within onTouchEvent. One way to fix this (already tested) is to call dispatchTouchEvent with a ACTION_CANCEL within setAdapter.  Possibly synching the two lines currently within setAdapter and the onTouchEvent() might also be needed, but probably just the dispatchTouchEvent will be enought to fix this problem.

**Comment by bauerca** [comment]
okay thanks. I'll get to this tomorrow. On the road...

**Comment by bauerca** [comment]
Could you send a traceback? thanks.

**Comment by leo-k** [comment]
Can't really send a traceback anymore. However, I can explain what happened and where. I used dslv within a ListFragment. The list would get refreshed, from a separate thread, this refresh was done through a setListAdapter call. If when this call was triggered the user was in the midsts of dragging a list item an exception would be thrown within onTouchEvent. This happened because, for some reason, the list went into some intermediate state where it had no items. Thus all the getChildAt calls would return null. Note that the setListAdapter that was being set had items in it, so I'm not sure what caused this intermediate step. However, the fix I described above solved this problem.
